kips/kip-1.md

```diff
@@ -227,7 +227,7 @@ contract UserPot is RBACWithAdmin, UserPotInterface, ERC165, UpgradeableInterfac
   }
 
   function calculateDeposit(address _user) public view returns (uint256) {
-      uint256 bal = 0;
+    uint256 bal = 0;
     address[] memory events = dataStore.getAddresses(_user, STORAGE_KEY_EVENTS);
     for (uint256 i = 0; i < events.length; i += 1) {
```

Comment:
Do we know the max number of events user can withdraw at one time without running out of gas (both `calculateDeposit` and `calculatePayout`)?

Reply:
Good question, basically it depends on loop cost. But this is something we'll only know during the implementation phase - at that point we can then internally limit the loop, thus allowing the user to call withdraw() as many times as is necessary.

However, due to how arrays work in solidity, note that in `updateUserData()` we've currently limited the array to max 10 items  = a user can be registered to attend at most 10 events at a time. This also limits the total gas required to execute updateUserData() and thus all dependent function calls. I believe we should calculate this maximum value (maybe it should be 20) such that it ensures we don't run out of gas but at the same time give the average user enough flexibility.

Reply:
So at least every 10 meeting people have to call `withdraw`? That may be a bit strange UX given this change is promoting users not to withdraw until they quit Kickback.

Reply:
No, we only store upto 10 _non-finished_ events at a time. The limitation is assuming the user isn't registered to attend more than 10 events at a time. 10 is arbitrary, we could use 20 or 100, depending on gast cost. Won't know until we calculate during implementation time.

Reply:
ah okay, it's not that bad then.